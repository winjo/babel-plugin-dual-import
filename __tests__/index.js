const pluginTester = require('babel-plugin-tester')
const createBabylonOptions = require('babylon-options')
const plugin = require('../index')
const babel = require('babel-core')
const dynamicSyntax = require('babel-plugin-syntax-dynamic-import')
const stage2 = require('babel-preset-stage-2')
const es2015 = require('babel-preset-es2015')

const babelOptions = {
  filename: 'currentFile.js',
  parserOpts: createBabylonOptions({
    stage: 2
  })
}

pluginTester({
  plugin,
  pluginOptions: { mode: 'eager' },
  babelOptions,
  snapshot: true,
  tests: {
    'static import': 'import("./Foo")',
    'static import (with relative paths)': 'import("../../Foo")',
    'static import (with file extension)': 'import("./Foo.js")',
    'static import (string template)': 'import(`./base`)',
    'static import (string template + relative paths)': 'import(`../../base`)',
    'dynamic import (string template)': 'import(`./base/${page}`)',
    'dynamic import (string template - dynamic at 1st segment)':
      'import(`./${page}`)',
    'dynamic import (string template + relative paths)':
      'import(`../../base/${page}`)'
  }
})

// toggle from test.skip to test.only when working on the plugin using Wallaby
test.skip('wallaby-live-coding', () => {
  const input =
    'import(`../async/${page}`);\n\nimport("../async/Foo.js");\n\nimport(`../async/page`);'
  // const input = 'import("../async/Foo.js")'

  const output = babel.transform(input, {
    filename: 'currentFile.js',
    plugins: [dynamicSyntax, [plugin, { mode: 'eager' }]],
    presets: [es2015, stage2]
  })

  output.code /*? */
})
