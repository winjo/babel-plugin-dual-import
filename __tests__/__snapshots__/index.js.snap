// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dynamic import (string template + relative paths) 1`] = `
"
import(\`../../base/\${page}\`)

      ↓ ↓ ↓ ↓ ↓ ↓

var _importCss = _interopRequireDefault(require(\\"babel-plugin-dual-import-with-mode/importCss.js\\")).default;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

Promise.all([import( /* webpackChunkName: 'base/[request]' */ /* webpackMode: 'eager' */\`../../base/\${page}\`), _importCss(\`base/\${page}\`)]).then(proms => proms[0]);
"
`;

exports[`dynamic import (string template - dynamic at 1st segment) 1`] = `
"
import(\`./\${page}\`)

      ↓ ↓ ↓ ↓ ↓ ↓

var _importCss = _interopRequireDefault(require(\\"babel-plugin-dual-import-with-mode/importCss.js\\")).default;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

Promise.all([import( /* webpackChunkName: '[request]' */ /* webpackMode: 'eager' */\`./\${page}\`), _importCss(\`\${page}\`)]).then(proms => proms[0]);
"
`;

exports[`dynamic import (string template) 1`] = `
"
import(\`./base/\${page}\`)

      ↓ ↓ ↓ ↓ ↓ ↓

var _importCss = _interopRequireDefault(require(\\"babel-plugin-dual-import-with-mode/importCss.js\\")).default;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

Promise.all([import( /* webpackChunkName: 'base/[request]' */ /* webpackMode: 'eager' */\`./base/\${page}\`), _importCss(\`base/\${page}\`)]).then(proms => proms[0]);
"
`;

exports[`static import (string template + relative paths) 1`] = `
"
import(\`../../base\`)

      ↓ ↓ ↓ ↓ ↓ ↓

var _importCss = _interopRequireDefault(require(\\"babel-plugin-dual-import-with-mode/importCss.js\\")).default;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

Promise.all([import( /* webpackChunkName: 'base' */ /* webpackMode: 'eager' */\`../../base\`), _importCss(\`base\`)]).then(proms => proms[0]);
"
`;

exports[`static import (string template) 1`] = `
"
import(\`./base\`)

      ↓ ↓ ↓ ↓ ↓ ↓

var _importCss = _interopRequireDefault(require(\\"babel-plugin-dual-import-with-mode/importCss.js\\")).default;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

Promise.all([import( /* webpackChunkName: 'base' */ /* webpackMode: 'eager' */\`./base\`), _importCss(\`base\`)]).then(proms => proms[0]);
"
`;

exports[`static import (with file extension) 1`] = `
"
import(\\"./Foo.js\\")

      ↓ ↓ ↓ ↓ ↓ ↓

var _importCss = _interopRequireDefault(require(\\"babel-plugin-dual-import-with-mode/importCss.js\\")).default;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

Promise.all([import( /* webpackChunkName: 'Foo' */ /* webpackMode: 'eager' */\\"./Foo.js\\"), _importCss(\\"Foo\\")]).then(proms => proms[0]);
"
`;

exports[`static import (with relative paths) 1`] = `
"
import(\\"../../Foo\\")

      ↓ ↓ ↓ ↓ ↓ ↓

var _importCss = _interopRequireDefault(require(\\"babel-plugin-dual-import-with-mode/importCss.js\\")).default;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

Promise.all([import( /* webpackChunkName: 'Foo' */ /* webpackMode: 'eager' */\\"../../Foo\\"), _importCss(\\"Foo\\")]).then(proms => proms[0]);
"
`;

exports[`static import 1`] = `
"
import(\\"./Foo\\")

      ↓ ↓ ↓ ↓ ↓ ↓

var _importCss = _interopRequireDefault(require(\\"babel-plugin-dual-import-with-mode/importCss.js\\")).default;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

Promise.all([import( /* webpackChunkName: 'Foo' */ /* webpackMode: 'eager' */\\"./Foo\\"), _importCss(\\"Foo\\")]).then(proms => proms[0]);
"
`;
